src/JuMP.jl

```diff
@@ -308,16 +285,34 @@ backend(model::Model) = model.moi_backend
 Return mode (Direct, Automatic, Manual) of model.
 """
 function mode(model::Model)
-    if !(backend(model) isa MOI.Bridges.LazyBridgeOptimizer{<:MOIU.CachingOptimizer} ||
-         backend(model) isa MOIU.CachingOptimizer)
+    if !(backend(model) isa MOIU.CachingOptimizer)
         return Direct
-    elseif caching_optimizer(model).mode == MOIU.Automatic
+    elseif backend(model).mode == MOIU.Automatic
         return Automatic
     else
         return Manual
     end
 end
 
+"""
+    bridge_constraints(model::Model)
+
+Return a `Bool` indicating whether the model `model` is in manual or automatic
+mode, the optimizer is set and unsupported constraints are automatically bridged
+to equivalent supported constraints when an appropriate transformation is
+available.
+"""
+function bridge_constraints(model::Model)
+    caching_optimizer = backend(model)
+    if caching_optimizer isa MOIU.CachingOptimizer
```

Comment:
Nit: `if backend(model) isa MOIU.CachingOptimizer`. It's weird call something `caching_optimizer` if it might not be one.

Reply:
I renamed it `moi_backend`. The advantage of storing it in a local variable is that inference can use the fact that it is in the first clause of the `if` to infer that the variable is a CachingOptimizer and we loose a bit of the disadvantage that `moi_backend` is not concretely typed